pkg/scaffold/resource/resource.go

```diff
@@ -78,33 +81,29 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
 	}
 
-	// todo: move it for the proper place since they are not validations and then, should not be here
-	// Add in r.Resource the Kind plural
-	if len(r.Resource) == 0 {
-		r.Resource = flect.Pluralize(strings.ToLower(r.Kind))
-	}
-	// Replace the caracter "-" for "" to allow scaffold the go imports
-	r.GroupImportSafe = strings.Replace(r.Group, "-", "", -1)
-    r.GroupImportSafe = strings.Replace(r.GroupImportSafe, ".", "", -1)
 	return nil
 }
 
-// isKindEmpty will return true if the --kind flag do not be informed
-// NOTE: required check if the flags are assuming the other flags as value
-func (r *Resource) isKindEmpty() bool {
-	return len(r.Kind) == 0 || r.Kind == "--group" || r.Kind == "--version"
+// isEmpty will return true if the provided flag is not set or the value is missing.
+// If provided without value it may take another flag as value, so we need to
+// check if it starts by a dash.
+func (_ *Resource) isEmpty(value string) bool {
+	return len(value) == 0 || strings.HasPrefix(value, "-")
```

Comment:
Note that it was not made generic in order to verify the specific values. We should not consider a text with the prefix `-` as empty since it can be a valid value for the k8s API. However, may you could verify here all possibilities `--group`, `--version`, `--kind`. However, the name of the func should be meaningful for it.

Reply:
I will check this but I think none of these 3 values can start by a dash. For example versions need to start with `v`. The benefit of checking it like this is that it will be future proof in case we add a new flag.
About the naming, it is a non exported method and its documentation clearly specifies that it checks flags, but if some alternative name such as `isGVKEmpty` feels more natural I can replace it.

Reply:
So, my point is: we cannot block more scenarios. We can just consider as empty the specific values that group, version and kind can assume when they are not properly filled.

Reply:
Checked it. `Group`s are validated to be a DNS1123 subdomain (that need to start with either a lowercase letter or a digit), `Version`s are validated against a regex that enforces first character to be `v`, and `Kind`s are checked against their PascalCase version which removes dashes. So none of then can start by dashes.

Reply:
And what happens with `kubebuilder create api --kind --controller --group example --version v1`?  Now we are giving as an error `Kind must be PascalCase (expected Controller was --controller)` which is the same issue you were solving but for other flags. There are other flags to be considered too, so the list will grow bigger. And in the future, we may add additional flags, like for example options for plugins, or `--builtin` to use the builtin apis. They are just example (and probably not that useful), but the point is that we need to handle more flags right now and we will probably have to handle even more later. Having to add each new flag to these 3 `if`s is error prone (will be hard to remember to do this and hard to debug to find that it needs to be done). Thats why I went for a flag agnostic method that just checks if they are flags (== start by dash).

Reply:
Please, I am just trying that we cannot block anything that starts with `-` (`|| strings.HasPrefix(value, "-")`). Each attribute has in its own validation and in POV it should be kept in this way.  

The idea of isEmptyGroup,  isEmptyVersion and isEmptyKind is just because if we  ` create api --kind --group example --version v1` for example the value for Kind will be `--group`.  

> `happens with kubebuilder create api --kind --controller --group example --version v1?`

In this case, the --kind is not empty and its value is ` --controller ` then, the specific validation will check if it is or not valid for it.

Reply:
`--controller` is a valid flag. It avoids being asked if you want to scaffold a controller. So the error given should not be `"--controller" is not a valid kind` but `kind is empty`. And the same as `--controller` there are other flags that could be used and we would have a very long `||` concatenation for each `if`. And if new flags were added in the future, we would still have to come here and update the `if`s. You need to catch any flag, you can't only provide a list of flags that are being catched here.

Reply:
The best approach would check if the cobra lib has a way to do it. I could not find a solution using cobra.  The second approach ass you wish. 

My point in the review is:
- We cannot block any value which starts with `-` since it can be a valid value for the attribute in k8s API
- isEmpty has not the responsibly to check if the value is valid or not for each attribute.